src/NUnitEngine/nunit.engine/Runners/DirectTestRunner.cs

```diff
@@ -71,7 +71,22 @@ public override TestEngineResult Explore(TestFilter filter)
             var result = new TestEngineResult();
 
             foreach (IFrameworkDriver driver in _drivers)
-                result.Add(driver.Explore(filter.Text));
+            {
+                string driverResult;
+
+                try
+                {
+                    driverResult = driver.Explore(filter.Text);
+                }
+                catch (Exception ex)
+                {
+                    if (ex is NUnitEngineException)
+                        throw;
```

Comment:
That would certainly be cleaner, assuming it works on the Mono builds as well. My guess is that it will - the language stuff is usually quite up to date.